notebook_data_redirector/sync.py

```diff
@@ -11,13 +11,21 @@ def lambda_handler(event, context):
     ddb_table = common.get_ddb_table()
     box_client, _ = common.get_box_client()
     root_folder = box_client.folder(common.BOX_FOLDER_ID)
+    root_shared = common.is_box_file_public(root_folder)
 
     LOGGER.info("Checking files in Box")
     shared_file_ids = set()
     shared_filepaths = set()
     count = 0
-    for file in common.iterate_files(root_folder):
+    for file, shared in common.iterate_files(root_folder, shared=root_shared):
         count += 1
+        if (not common.is_box_file_public(file)) and shared:
+            # this includes an api call
+            if file.type == "file":
```

Comment:
Did you find a case where iterate_files yielded a non-file object?

Reply:
oh, hmm.. no, good point, I was just reading the code in isolation.

This does actually mean that something isn't handled. I thought this would unshare folders nested under a shared folder, but I guess it won't.